src/components/search.js

```diff
@@ -20,10 +20,59 @@ export function createResultElements(searchQueryResults) {
   searchQueryResults.forEach(searchQueryResult => {
     let randomBgColorClass =
       bgColorClasses[Math.floor(Math.random() * bgColorClasses.length)];
+
     let searchResultsEntry = createElement('div', {
       classList: 'search-results__entry ' + randomBgColorClass,
       innerText: searchQueryResult
     });
+
+    let favoriteButton = createElement('button', {
+      classList: 'search-results__favorite',
+      innerText: 'favorit'
+    });
+
+    /* Favorites functionality */
+
+    favoriteButton.dataset.entry = searchQueryResult;
```

Comment:
I don't think it's necessary to save the query results in the dataset of the button.
Any reason for that?

Reply:
Yes, search query result might be the wrong name.

Here is what I want to do: 
the favorite button should toggle favorite status for a Pokemon. When clicked, I need to access the name of the Pokemon to push it to the favorites array. So when the button is created, I add the name of the Pokemon to the dataset so I can access it easily.

I could also access the name of the Pokemon from the parent node but I think there will be future changes to the structure. Also I plan to move the favorites functionality to a module to make it usable on a pokemon detail view so I think the dataset might be useful for the functionality.